# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1348082890 14400
#      Wed Sep 19 15:28:10 2012 -0400
# Node ID 57e685d135ec727685b6f23a9f33c9c4f0294b8c
# Parent  cd6ce11b9c5733792ff108b882e11e7665d541aa
include oct-obj.h in generated builtin-defun-decls.h file

* mkbuiltins: Emit #include "oct-obj.h" statement instead of using a
forward declaration of octave_value_list.

diff --git a/libinterp/mkbuiltins b/libinterp/mkbuiltins
--- a/libinterp/mkbuiltins
+++ b/libinterp/mkbuiltins
@@ -45,22 +45,22 @@ esac
 if $make_header; then
 
   cat << \EOF
 // DO NOT EDIT!  Generated automatically by mkbuiltins.
 
 #if !defined (octave_builtin_defun_decls_h)
 #define octave_builtin_defun_decls_h 1
 
+#include "oct-obj.h"
+
 #define DEFUN_DECL(name) \
   extern OCTINTERP_API octave_value_list \
   name (const octave_value_list& = octave_value_list (), int = 0);
 
-class octave_value_list;
-
 EOF
 
  $SED -n -e 's/ *\(XDEFUN\|XDEFCONSTFUN\)_INTERNAL *( *\([_A-Za-z][_A-Za-z0-9]*\) *,.*$/DEFUN_DECL (F\2);/p; s/^ *XDEFUNX_INTERNAL *( *"\([_A-Za-z][_A-Za-z0-9]*\)" *,.*$/DEFUN_DECL (F\1);/p' "$@"
 
   cat << \EOF
 
 #undef DEFUN_DECL
 
